dll/win32/iernonce/dialog.cpp

```diff
@@ -37,10 +37,12 @@ BOOL ProgressDlg::RunDialogBox()
     // Show the dialog and run the items only when the list is not empty.
     if (m_RunOnceExInst.m_SectionList.GetSize() != 0)
     {
-        if (DoModal() == -1)
+        INT_PTR Ret = DoModal();
+        if (Ret == -1)
         {
             return FALSE;
         }
+        return Ret;
```

Comment:
The value I passed to `EndDialog` is always a BOOL (0 or 1), so I think it's ok...?

Reply:
Why not then just do:
```c
return (DoModal() != -1);
```

Reply:
if Ret is 0, it means DoModal succeeded but we passed to EndDialog `0`, something's wrong
but I think it can be simplified to `return (DoModal() == 1);` , I will apply it

Reply:
Just check what the DoModal function returns when everything succeeds.